### PR TITLE
feat(disclosure): add element customization

### PR DESCRIPTION
**Pull request by gloriliale**
Added customization capability to the disclosure package. Lingering question of why testing for backgroundColor doesn't seem to be working, which is commented out of the test file.

### CODE REVIEW

File: packages/paste-core/components/disclosure/__tests__/disclosure.test.tsx [hunk 1]
```diff
@@ -164,6 +195,87 @@ describe('Disclosure', () => {
...
+
+    it('should set a custom element data attribute for custom named Disclosure components', () => {
+      render(
+        <Theme.Provider theme="default">
+          <MockCustomElementDisclosure />
+        </Theme.Provider>
+      );
+
+      const renderedDisclosureHeading = screen.getByTestId('disclosure-heading');
+      const renderedDisclosure = renderedDisclosureHeading.parentElement.parentElement;
```
Comment (TheSisb) [inline packages/paste-core/components/disclosure/__tests__/disclosure.test.tsx:223]: It may be better to give this a testid

**Review by TheSisb — APPROVED** [review]

File: packages/paste-core/components/disclosure/__tests__/disclosure.test.tsx (hunk 1 above)
Comment (gloriliale) [reply packages/paste-core/components/disclosure/__tests__/disclosure.test.tsx:223]: I tried to, put we don't spread BoxProps on the Disclosure prop, so it doesn't show up. I should've asked if that's on purpose or not actually. I assumed that it was for a specific reason, and this is my workaround for grabbing Disclosure. If not, I could just do safelySpreadBoxProps onto Disclosure and then use testId.

**Review by nkrantz — APPROVED** [review]
Lookin goooood! I think we have been adding 3 examples to the customization stories, one normal, one customized, and one customized with a custom `element`. Not sure if that's necessary, but just wanted to call it out in case.

Also, I also have this written down on my list, but just want to make a note that once the Test Theme is merged in, we'll have to go back and update the Disclosure and Help Text tests to use it.

File: packages/paste-core/components/disclosure/__tests__/disclosure.test.tsx (hunk 1 above)
Comment (gloriliale) [reply packages/paste-core/components/disclosure/__tests__/disclosure.test.tsx]: Talked with Simon about it and implemented the the latter option (use safelySpreadBoxProps on Disclosure and getByTestId in tests). Thanks for pushing on this, @TheSisb!

File: packages/paste-core/components/disclosure/src/DisclosureHeading.tsx [hunk 2]
```diff
@@ -56,6 +60,7 @@ const StyledDisclosureHeading = React.forwardRef<HTMLDivElement, StyledDisclosur
         >
           <Box
             as="span"
+            element={`${element}_ICON`}
```
Comment (SiTaggart) [inline packages/paste-core/components/disclosure/src/DisclosureHeading.tsx:63]: There's a story for icon (which is weird if you had to add this @TheSisb, was it actually working?) but no test to make sure we don't accidentally remove it.

Can we add a test for DISCLOSURE_HEADING_ICON?

**Review by SiTaggart — CHANGES_REQUESTED** [review]

File: packages/paste-core/components/disclosure/src/DisclosureHeading.tsx (hunk 2 above)
Comment (TheSisb) [reply packages/paste-core/components/disclosure/src/DisclosureHeading.tsx:63]: I just added the story for icon, there wasn't one before.
I'll add a test.

**Review by SiTaggart — APPROVED** [review]

**Pull request merged**